src/types/types.ts

```diff
@@ -270,6 +292,32 @@ export type CallJobMetadata = {
     rec_id?: string;
 };
 
+export function isCallJobMetadata(obj: unknown): obj is CallJobMetadata {
+    if (typeof obj !== 'object' || obj === null) {
+        return false;
+    }
+
+    const metadata = obj as CallJobMetadata;
+
+    if (typeof metadata.file_id !== 'string') {
+        return false;
+    }
+
+    if (typeof metadata.post_id !== 'string') {
+        return false;
+    }
+
+    if (metadata.tr_id !== undefined && typeof metadata.tr_id !== 'string') {
```

Comment:
Is the `undefined` check necessary?

Reply:
I don't think so...

Reply:
The reason to use `undefined` is because `tr_id` is optional. So what we are doing here is saying...
- Is there a value on `tr_id`? (`tr_id !== undefined`)
- If so, is the value a string? (`typeof tr_id === 'string'`)

If we don't check for undefined, we have the following things that may happen:
- Only check `typeof tr_id === 'string'`
   - An object that doesn't have defined `tr_id` won't pass the check
- `tr_id && typeof tr_id === 'string'`
   - An object that `tr_id` is a 0, a null or any other falsy value would be considered valid. This probably have no implications, since you usually check whether it is defined by checking whether the value is truthy or falsy, but this direction is more correct.

Reply:
Thanks. Could we add exceptions to make the linter happy?